docs: documentation overhaul (#1651)
GIScience/openrouteservice: docs/api-reference/endpoints/directions/routing-options.md

# Routing Options

With the request body parameter `options`, advanced routing options can be specified for a directions request.

> The `options` parameter is a JSON map, multiple of the here mentioned parameters can be set. 

Reviewer: map -> object throughout the whole file

> # Routing Options

Reviewer 2: This is great, we should probably link to this from the API Playground? @TheGreatRefrigerator

| Value          | Description                                  |
|----------------|----------------------------------------------|
| `"all"`        | for no border crossing                       |
| `"controlled"` | cross open borders but avoid controlled ones | 

### `avoid_countries` 
An integer array of country ids to exclude from routing with **`driving-*`** profiles. Can be used together with `"avoid_borders": "controlled"`. 
The list of countries and application examples can be found in the [country list](../../../technical-details/country-list.md).

`"[11,193]"` would exclude Austria and Switzerland. 

### `avoid_features`
A string array of features to avoid. The available features are :

| Feature    | Available for                               |
|------------|---------------------------------------------|
| `highways` | driving-\*                                  |
| `tollways` | driving-\*                                  |
| `ferries`  | driving-\*, cycling-\*, foot-\*, wheelchair |
| `fords`    | driving-\*, cycling-\*, foot-\*             |
| `steps`    | cycling-\*, foot-\*, wheelchair             |

### `avoid_polygons` 

> Comprises areas to be avoided for the route. Formatted as [geojson polygon](http://geojson.org/geojson-spec.html#id4) or [geojson multipolygon](http://geojson.org/geojson-spec.html#id7).

Reviewer 2: We should link to RFC 7946 here.

```suggestion
Comprises areas to be avoided for the route. Formatted as [geojson polygon](https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.3) or [geojson multipolygon](https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.6).
```

  | `height`   | Number  | Height restriction in meters.                                                                                                     |
  | `axleload` | Number  | Axleload restriction in tons.                                                                                                     |
  | `weight`   | Number  | Weight restriction in tons.                                                                                                       |
  | `hazmat`   | Boolean | Specifies whether to use appropriate routing for delivering hazardous goods and avoiding water protected areas. Default is false. |

* for `wheelchair`:

  | Parameter             | Type    | Description                                                                                                                 |
  |-----------------------|---------|-----------------------------------------------------------------------------------------------------------------------------|
  | `surface_type`        | String  | Specifies the minimum [surface type](http://wiki.openstreetmap.org/wiki/Key:surface). Default is `"cobblestone:flattened"`. |

>   | `track_type`          | String  | Specifies the minimum [grade](http://wiki.openstreetmap.org/wiki/Key:tracktype) of the route. Default is `"grade1"`.        |

Reviewer 2: Especially since we're using `gradient` above, people may confuse this with the incline.
```suggestion
  | `track_type`          | String  | Specifies the minimum [quality](http://wiki.openstreetmap.org/wiki/Key:tracktype) of the route. Default is `"grade1"`.        |
```
Maybe something like *surface firmness* might even be better, the OSM wiki may provide inspiration ;)

  | `smoothness_type`     | String  | Specifies the minimum [smoothness](http://wiki.openstreetmap.org/wiki/Key:smoothness) of the route. Default is `"good"`.    |
  | `maximum_sloped_kerb` | Number  | Specifies the maximum height of the sloped kerb in meters. Values are `0.03`, `0.06`(default), `0.1` or `any`.              |
  | `maximum_incline`     | Integer | Specifies the maximum incline as a percentage. `3`, `6`(default), `10`, `15` or `any`.                                      |
  | `minimum_width`       | Number  | Specifies the minimum width of a road in meters.                                                                            |

### `round-trip` 

> [//]: # (TODO: write)

Reviewer 2: As it says, ToDo :D

Reviewer 3: added something, @koebi

### `vehicle_type`
For `profile=driving-hgv` only. It is needed for **vehicle restrictions** to work. Possible values:

* `hgv`
* `bus`
* `agricultural`
* `delivery`
* `forestry` 

> * `goods`

Reviewer 2: This could use info on what's what, or links to the OSM wiki?

Reviewer: @koebi I've linked to the tag filtering instead, where it is described to which osm keys the values are connected

> ## Examples

Reviewer 2: We should probably check the examples to make sure they work.

Reviewer: Done

HINT: *If your request works without the `options` object, but returns an error _with_ it: try to [URL-encode](#url-encoding) the options object!*

Some options examples in readable and minified JSON form:

> for `profile=driving-car`:

Reviewer 2: ```suggestion
### for `profile=driving-car`:
```

```json
{
    "avoid_features": ["ferries", "tollways"]
}
```
`{"avoid_features":["ferries","tollways"]}`

> for `profile=cycling-*`:

Reviewer 2: ```suggestion
### for `profile=cycling-*`:
```

```json
{
  "avoid_features": ["steps"],
  "profile_params": {
      "weightings": {
          "steepness_difficulty": 2
      }
  },
  "avoid_polygons": {
      "type": "Polygon",
      "coordinates": [
          [ [100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0] ]
   ]}
}
```
`{"avoid_features":["steps"],"profile_params":{"weightings":{"steepness_difficulty":2}}}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`

> for `profile=foot-*`:

Reviewer 2: ```suggestion
### for `profile=foot-*`:
```

```json
{
    "avoid_features": ["fords","ferries"],
    "profile_params": {
        "weightings": {
            "green": {
                "factor": 0.8
              },
            "quiet": {
                "factor": 1.0
            }
        }
    },
    "avoid_polygons": {  
        "type": "Polygon",
        "coordinates": [
            [ [100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0] ]
     ]}
}
```
`{"avoid_features":["fords","ferries"],"profile_params":{"weightings":{"green":{"factor":0.8},"quiet":{"factor":1.0}}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`

> for `profile=driving-hgv`:

Reviewer 2: ```suggestion
### for `profile=driving-hgv`:
```

```json
{
    "avoid_features": ["ferries","tollways"],
    "vehcile_type": "hgv",
    "profile_params": {
        "restrictions": {
            "length": 30,
            "width": 30,
            "height": 3,
            "axleload": 4,
            "weight": 3,
            "hazmat": true
        }
    },
    "avoid_polygons": {  
        "type": "Polygon",
        "coordinates": [
            [ [100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0] ]
     ]}
}
```
`{"avoid_features":["ferries","tollways"],"vehicle_type":"hgv","profile_params":{"restrictions":{"length":30,"width":30,"height":3,"axleload":4,"weight":3,"hazmat":true}},"avoid_polygons":{"type":"Polygon","coordinates":[[[100.0,0.0],[101.0,0.0],[101.0,1.0],[100.0,1.0],[100.0,0.0]]]}}`

> for `profile=wheelchair`:

Reviewer 2: ```suggestion
### for `profile=wheelchair`:
```
